velox/functions/prestosql/types/TimestampWithTimeZoneType.h

```diff
@@ -54,9 +54,8 @@ class TimestampWithTimeZoneCastOperator : public exec::CastOperator {
 
 /// Represents timestamp with time zone as a number of milliseconds since epoch
 /// and time zone ID.
-class TimestampWithTimeZoneType : public RowType {
```

Comment:
As mentioned in the discussion https://github.com/facebookincubator/velox/discussions/2511,
I feel we should have the same precision for Timestamp and TimestampTz. That means we should make this an extension of the `Timestamp` physical type.

Reply:
Hi @majetideepak Thank you for your reply.
> I feel we should have the same precision for Timestamp and TimestampTz

I agree with this. But I have a question, now TimestampTz is a Presto-specific type, and Presto's TimestampTz is millisecond precision. I see that Spark also supports TimestampTz, and the precision seems to be microseconds. If we want TimestampTz to have the same precision as Timestamp, do we have to move TimestampTz to a Velox type instead of being a Presto-specific type?


> That means we should make this an extension of the Timestamp physical type.

If TimestampTz extends from the Timestamp physical type, then how do we store timeZone?

Reply:
I found this blog helpful. https://www.databricks.com/blog/2020/07/22/a-comprehensive-look-at-dates-and-timestamps-in-apache-spark-3-0.html
There are three types of timestamp types. 
`Timestamp Without Time Zone`, `Timestamp with Time Zone`, `Timestamp with Session Time Zone`. 
Spark/Parquet support the third type. I think we can support both `Timestamp Without Time Zone` and `Timestamp with Session Time Zone` using the regular timestamp Type in Velox.
We need a `Timestamp with Time Zone` type to match Presto spec.

Reply:
I read the above article, and my understanding is: `Timestamp with Time Zone` in the article is actually similar to the implementation of this PR; `Timestamp Without Time Zone` is similar to Velox’s `Timestamp`; `Timestamp With Session Time Zone` needs to take into account the local session Time Zone, and there is no corresponding implementation in Velox. I don't know if I understand it correctly.

Reply:
Your understanding is correct. I don't think we need a special type for `Timestamp with Session Time Zone` in Velox. We can use the Timestamp column after adjusting the values to the session time zone.
Parquet and DWRF both specify in their metadata if a timestamp column is UTC adjusted or not. If it is, we use the session time zone to convert the values and store them in the Timestamp column.
https://github.com/apache/parquet-format/blob/master/LogicalTypes.md